WordPressAuthenticator/Email Client Picker/LinkMailPresenter.swift

```diff
@@ -0,0 +1,39 @@
+import MessageUI
+
+
+/// Email picker presenter
+class LinkMailPresenter {
+    /// Presents the available mail clients in an action sheet. If none is available,
+    /// Falls back to Apple Mail and opens it.
+    /// If not even Apple Mail is available, presents an alert to check your email
+    /// - Parameters:
+    ///   - viewController: the UIViewController that will present the action sheet
+    ///   - appSelector: the app picker that contains the available clients. Nil if no clients are available
+    ///                  reads the supported email clients from EmailClients.plist
+    func presentEmailClients(on viewController: UIViewController,
+                             appSelector: AppSelector? = AppSelector()) {
+
+        guard let picker = appSelector else {
+            // fall back to Apple Mail if no other clients are installed
+            if MFMailComposeViewController.canSendMail(), let url = URL(string: "message://") {
+                UIApplication.shared.open(url)
+            } else {
+                showAlertToCheckEmail(on: viewController)
+            }
+            return
+        }
+        viewController.present(picker.alertController, animated: true)
+    }
+
+    private func showAlertToCheckEmail(on viewController: UIViewController) {
+        let title = NSLocalizedString("Please check your email", comment: "Alert title for check your email during logIn/signUp.")
+        let message = NSLocalizedString("Please open your email app and look for an email from WordPress.com.", comment: "Message to ask the user to check their email and look for a WordPress.com email.")
```

Comment:
I know these were existing strings, but perhaps we should match them to the messages we show on the web:

Title: "Check your email!"
Message: "We just emailed a link to EMAIL_ADDRESS. Please check your mail app and tap the link to log in."

If it's too difficult to include the email address here, we could just change that to "We just emailed you a link to log in. Please check your mail app and tap the link."

Reply:
we can use the email address. I think it's safe to assume that the address will always be a valid email address (otherwise users would not get a magic link and not land to this page). Do you see any issue with this assumption?

Reply:
I don't see an issue – it informs the user where we've attempted to send a link, so if it's not a valid email address for some reason it should help the user notice that.